pkgs/tools/networking/v2ray/default.nix

```
@@ -0,0 +1,35 @@
+{ lib, fetchFromGitHub, buildGoPackage, go }:
+
+buildGoPackage rec {
+  pname = "v2ray";
+  version = "4.20.0";
+
+  src = fetchFromGitHub {
+    owner = "V2Ray";
+    repo = "v2ray-core";
+    rev = "v${version}";
+    sha256 = "18d4zk9z3080vj0whashzb0jdxz0xlpk40vy1wmr1hb163zp0b7z";
+  };
+
+  goPackagePath = "v2ray.com/core";
+  goDeps = ./deps.nix;
+
+  nativeBuildInputs = [ go ];
+
+  buildPhase = ''
+    runHook preBuild
+    go build -o v2ray v2ray.com/core/main
+    go build -o v2ctl v2ray.com/core/infra/control/main
+  '';
+
+  installPhase = ''
+    mkdir -p $bin/bin
+    mv v2ray v2ctl $bin/bin
```

if you want, both lines can be shortened to:
```suggestion
    install -Dm755 v2ray v2ctl -t $bin/bin
```